portfolio/src/main/webapp/script.js

```diff
@@ -12,6 +12,47 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/**
+ * Randomly selects a background color when page refreshes.
+ */
+function setBackgroundColor() {
```

Comment:
This function name tells me that it changes background color but it also changes text color. I recommend some more function decomposition.

At a high level you could have setPageColors(). Maybe it could look something like this,

```
function setPageColors() {
  const bgColors = ...;
  const bgColor = ...;

  document.body.style.backgroundColor = bgColor;

  textColor = getContrastColor(bgColor);
  document.body.style.color = textColor;
}

function getContrastColor(color) {
  ...
}
```

What do you think?